src/client/datascience/notebook/notebookSerializer.ts

```diff
@@ -76,8 +76,9 @@ export class NotebookSerializer implements VSCNotebookSerializer {
         return Buffer.from(this.serialize(data), 'utf-8');
     }
     private serialize(data: NotebookDocument | NotebookData): string {
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        const notebookContent: Partial<nbformat.INotebookContent> = (data.metadata?.custom as any) || {};
+        const notebookContent: Partial<nbformat.INotebookContent> =
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            JSON.parse(JSON.stringify(data.metadata?.custom as any)) || {};
```

Comment:
This is the where the bug is, we need to ensure we clone it. VS Code freezes the metadata (making it readonly).